augur/index.py

```diff
@@ -9,14 +9,50 @@
 import csv
 
 from .io import open_file, read_sequences
+from .utils import is_vcf, read_vcf
 
 
 def register_arguments(parser):
-    parser.add_argument('--sequences', '-s', required=True, help="sequences in fasta format")
+    parser.add_argument('--sequences', '-s', required=True, help="sequences in FASTA or VCF formats. Augur will summarize the content of FASTA sequences and only report the names of strains found in a given VCF.")
     parser.add_argument('--output', '-o', help="tab-delimited file containing the number of bases per sequence in the given file. Output columns include strain, length, and counts for A, C, G, T, N, other valid IUPAC characters, ambiguous characters ('?' and '-'), and other invalid characters.", required=True)
     parser.add_argument('--verbose', '-v', action="store_true", help="print index statistics to stdout")
 
 
+def index_vcf(vcf_path, index_path):
```

Comment:
Consider moving this into `filter.py` as an internal function instead of exposing this as an index. If we want to support indexing VCFs as an Augur feature that others would use, we should keep this here and eventually implement the full index output format.

Feedback from @emmahodcroft and @cassiawag about the value of this functionality would be helpful.

Reply:
I'm very interested in augur support for vcfs, but I don't have any understanding of the current status of the codebase.